strawberryfields/api/connection.py

```diff
@@ -227,7 +227,15 @@ def get_job_result(self, job_id: str) -> Result:
             with io.BytesIO() as buf:
                 buf.write(response.content)
                 buf.seek(0)
+
                 samples = np.load(buf, allow_pickle=False)
+
+                if samples.dtype.type is np.int8:
+                    # Samples represent photon numbers.
+                    # Convert to int64, to avoid unexpected behaviour
+                    # when users postprocess these samples.
```

Comment:
Curious about this one: what type of errors arose for the users? :slightly_smiling_face:

Reply:
Users were performing numerical post-processing on the returned samples, and hitting overflow 🙂 

```python
>>> samples = np.array([1, 0, 2, 7], dtype=np.int8)
>>> samples**6
array([   1,    0,   64, -111], dtype=int8)
```

Reply:
Great catch! :blush: